src/server/admin/server/api_server.go

```diff
@@ -257,6 +263,87 @@ func (a *apiServer) Extract(request *admin.ExtractRequest, extractServer admin.A
 			}
 		}
 	}
+	if !request.NoEnterprise {
+		state, err := pachClient.Enterprise.GetState(pachClient.Ctx(), &enterprise.GetStateRequest{})
+		if err != nil {
+			return err
+		}
+		// Don't write out expired or missing enterprise keys
+		if state.State == enterprise.State_ACTIVE {
+			if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{ActivateEnterprise: &enterprise.ActivateRequest{ActivationCode: state.ActivationCode}}}); err != nil {
+				return err
+			}
+		} else {
+			logrus.Warnf("Enterprise license state: %v, not extracting", state.State)
+		}
+	}
+
+	if !request.NoAuth {
+		config, err := pachClient.GetConfiguration(pachClient.Ctx(), &auth.GetConfigurationRequest{})
+		if err == nil {
+			// Activate authentication as `robot:extractrestore`. This token will be used by the Restore process for subsequent operations.
+			if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{ActivateAuth: &auth.ActivateRequest{Subject: "robot:extractrestore"}}}); err != nil {
+				return err
+			}
+
+			// Extract all robot tokens. This includes the backup root token created by pachctl, which can be used to recover the cluster if the IDP configuration is wrong.
+			robotTokens, err := pachClient.ExtractAuthTokens(pachClient.Ctx(), &auth.ExtractAuthTokensRequest{})
+			if err != nil {
+				return err
+			}
+
+			for _, t := range robotTokens.Tokens {
+				if err := writeOp(&admin.Op{Op1_12: &admin.Op1_12{RestoreAuthToken: &auth.RestoreAuthTokenRequest{Token: t}}}); err != nil {
+					if !auth.IsErrExpiredToken(err) {
```

Comment:
We have to do some gross mapping of TTLs to absolute timestamps, and it's possible for an extract to contain an expired robot token. In that case we just log and move on rather than blocking the restore.